op-e2e/actions/altda/altda_test.go

```diff
@@ -607,3 +643,42 @@ func TestAltDA_Finalization(gt *testing.T) {
 	// given 12s l1 time and 1s l2 time, l2 should be 12 * 3 = 36 blocks finalized
 	require.Equal(t, uint64(36), a.sequencer.SyncStatus().FinalizedL2.Number)
 }
+
+func TestAltDA_FinalizationAfterEthDAFailover(gt *testing.T) {
```

Comment:
Nice, this test case covers ethDA -> altDA. Can we also have a similar case ethDA -> altDA -> ethDA, simulating a temp altDA failure?

Reply:
Done: [ce0267b](https://github.com/ethereum-optimism/optimism/pull/12845/commits/ce0267b8fd1576680e1ece4aa36541f5116347a3)
I don't fully understand the finalization behavior. Look at the weird `if i==0` cases. Would appreciate help figuring out whether this behavior is really what we want, and if so, how to best explain it in the comment/assert logic.